test/js-parser.html

```diff
@@ -202,48 +202,56 @@
     suite('behavior metadata', function() {
 
       var parsed, byName;
+      var analyzer;
       setup(function(done) {
-        var loader = new hyd.Loader();
-        var resolver = new hyd.XHRResolver();
-        loader.addResolver(resolver);
-        loader.request("static/js-behaviors.js").then(function(content){
-          parsed = hyd._jsParse(content);
-          byName = {};
-          parsed.behaviors.forEach(function(behavior) {
-            byName[behavior.is] = behavior;
-          });
+        hyd.Analyzer.analyze("static/html-behaviors.html").then(function(analyzed){
```

Comment:
With mocha you can just return a promise in a test rather than using done(). It will then wait on the promise to resolve successfully, much the same way that it now waits for done() to be called with no arguments. 

I bet this will work for setup as well.